### PR TITLE
LAA, LVer: add pre-commit tests for #96656

### PR DESCRIPTION
**Pull request by artagnon**
The issue is in LoopAccessAnalysis, but the regression was seen in the user LoopVersioning. Hence, add pre-commit tests for both, in preparation to fix the issue in LoopAccessAnalysis.

### CODE REVIEW

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

File: llvm/test/Analysis/LoopAccessAnalysis/pr96656.ll [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+; CHECK-NEXT:      SCEV assumptions:
+; CHECK-NEXT:      Equal predicate: %load == 1
+; CHECK-EMPTY:
+; CHECK-NEXT:      Expressions re-written:
+; CHECK-NEXT:      [PSE] %gep10 = getelementptr double, ptr %gep8, i64 %mul:
+; CHECK-NEXT:        {(8 + %arg1),+,(8 * (sext i32 %load to i64))<nsw>}<%loop.body>
+; CHECK-NEXT:        --> {(8 + %arg1),+,8}<%loop.body>
+;
+entry:
+  %load = load i32, ptr %arg, align 4
```
[the remark this answers is not in the document]
Comment (artagnon) [reply llvm/test/Analysis/LoopAccessAnalysis/pr96656.ll:22]: Further down, there is a `store double %load5, ptr %arg, align 8`, so this is not possible.

[the remark this answers is not in the document]
Comment (fhahn) [reply llvm/test/Analysis/LoopAccessAnalysis/pr96656.ll:22]: Ah right, but that could be replaced by a `ret double %load5`?

[the remark this answers is not in the document]
Comment (artagnon) [reply llvm/test/Analysis/LoopAccessAnalysis/pr96656.ll:22]: Sorry, merged before I saw your comment. Investigating now.

[the remark this answers is not in the document]
Comment (artagnon) [reply llvm/test/Analysis/LoopAccessAnalysis/pr96656.ll:22]: Okay, so it requires non-trivial changes for the `loop.exit` bb, and the predicate would change to `%arg == 1`, which I think is an entirely different test from speculating on the load.

**Pull request merged**